src/main/java/com/github/firmwehr/gentle/backend/ir/codegen/CodeSelection.java

```diff
@@ -386,6 +390,22 @@ public void visit(Store node) {
 		block.nodes().add(ikeaMovStore);
 	}
 
+	@Override
+	public void visit(Shr node) {
+		IkeaBløck block = blocks.get((Block) node.getBlock());
```

Comment:
git didn't pick that up I guess, but code selection requires each visit method to check if the node has been replace, once you rebase on master. Just a heads up.

Reply:
rebased and updated, please check if everything is correct